src/main/java/picard/illumina/parser/TileMetricsUtil.java

```diff
@@ -51,33 +54,76 @@
  * @author mccowan
  */
 public class TileMetricsUtil {
-    /** The path to the directory containing the tile metrics file relative to the basecalling directory. */
+
+    private static final Log log = Log.getInstance(TileMetricsUtil.class);
+
+    public static String CYCLE_TWENTY_FIVE_NAME = "C25.1";
+    /**
+     * The path to the directory containing the tile metrics file relative to the basecalling directory.
+     */
     public static String INTEROP_SUBDIRECTORY_NAME = "InterOp";
-    
-    /** The expected name of the tile metrics output file. */
+
+    /**
+     * The expected name of the tile metrics output file.
+     */
     public static String TILE_METRICS_OUT_FILE_NAME = "TileMetricsOut.bin";
 
     private final static Log LOG = Log.getInstance(TileMetricsUtil.class);
 
-    /** Returns the path to the TileMetrics file given the basecalling directory. */
-    public static File renderTileMetricsFileFromBasecallingDirectory(final File illuminaRunDirectory) {
-        return new File(new File(illuminaRunDirectory, INTEROP_SUBDIRECTORY_NAME), TILE_METRICS_OUT_FILE_NAME);
+    /**
+     * Returns the path to the TileMetrics file given the basecalling directory.
+     */
+    public static File renderTileMetricsFileFromBasecallingDirectory(final File illuminaRunDirectory, boolean isNovaSeq) {
+        if (isNovaSeq) {
+            return new File(new File(illuminaRunDirectory, INTEROP_SUBDIRECTORY_NAME + File.separator + CYCLE_TWENTY_FIVE_NAME), TILE_METRICS_OUT_FILE_NAME);
+        } else {
+            return new File(new File(illuminaRunDirectory, INTEROP_SUBDIRECTORY_NAME), TILE_METRICS_OUT_FILE_NAME);
+        }
+    }
+
+    public static Collection<Tile> parseTileMetrics(final File tileMetricsOutFile,
+                                                    final Map<Integer, File> phasingMetricsFiles,
+                                                    final ReadStructure readStructure,
+                                                    final ValidationStringency validationStringency)
+            throws FileNotFoundException {
+        TileMetricsOutReader tileMetricsIterator = new TileMetricsOutReader
+                (tileMetricsOutFile, 3);
+        final Collection<IlluminaTileMetrics> tileMetrics = determineLastValueForLaneTileMetricsCode(tileMetricsIterator);
+        final Map<String, ? extends Collection<IlluminaTileMetrics>> locationToMetricsMap = partitionTileMetricsByLocation(tileMetrics);
+        final Collection<Tile> tiles = new LinkedList<>();
+        for (final Map.Entry<String, ? extends Collection<IlluminaTileMetrics>> entry : locationToMetricsMap.entrySet()) {
```

Comment:
Can you eliminate the ? here by making partitionTileMetricsByLocation() a generic function? What other types are expected here?